org.uqbar.project.wollok/src/org/uqbar/project/wollok/scoping/WollokGlobalScopeProvider.xtend

```diff
@@ -114,12 +114,20 @@ class WollokGlobalScopeProvider extends DefaultGlobalScopeProvider {
 	 */
 	def loadResource(URI uri, ResourceSet resourceSet) {
 		try {
-			val resource = resourceSet.getResource(uri, true)
-			resource.load(#{})
-			resourceDescriptionManager.getResourceDescription(resource).exportedObjects
+			var Iterable<IEObjectDescription> exportedObjects
+			//checkResourceSet(resourceSet as XtextResourceSet)
+			exportedObjects = WollokResourceCache.getResource(uri)
+			if (exportedObjects == null) {
+				val resource = resourceSet.getResource(uri, true)
+				resource.load(#{})
+				exportedObjects = resourceDescriptionManager.getResourceDescription(resource).exportedObjects
+				WollokResourceCache.addResource(uri, exportedObjects)
+			}
+			exportedObjects
```

Comment:
This only applies to SDK files:

```scala
	static synchronized def void addResource(URI uri,  Iterable<IEObjectDescription> objects) {
		if (uri.toString.startsWith(CLASSPATH) && (getResource(uri) == null)) {
			resourceCache.put(uri, objects)
		}
	}
```

So I think there is no problem with that.